twilio-connector/twilio_connector.py

```diff
@@ -0,0 +1,145 @@
+    def add_message(self, conv_id, author, message):
+        """Add a new message in the conversation 
+        Args:
+            conv_id: string representing unique id of of the conversation
+            author: string representing the author of the message
+            message: string representing the body of the message
+        Returns:
+            
+        """
+        self.conversation = self.client.conversations \
+                .v1 \
+                .conversations(conv_id) \
+                .messages \
+                .create(author=author, body=message)
+
+        return None
+
+    @sleep_and_retry
+    @limits(calls=NUM_OF_CALLS, period=TIME_PERIOD_SEC)
+    def get_conversation(self, conv_id, leng_limit=None):
+        """Pulls a conversation messages using a unique conversation id from most recent.
+        
+        Makes an API call to the conversations end point and pulls a single conversation using it's unique conversation id.
+        Follows rate limits set by Twilio.
+        Args:
+            conv_id: string representing unique id for a conversation.
+            leng_limit: limit the number of messages fetched from the most recent
+        Returns:
+            messages: list of conversation message instances
+        """
+        if leng_limit is not None and leng_limit.isdigit():
+            leng_limit = int(leng_limit)
+            self.messages = self.client.conversations \
+                                .v1 \
+                                .conversations(conv_id) \
+                                .messages \
+                                .list(order='desc', limit=leng_limit)
+        else:
+            self.messages = self.client.conversations \
+                    .v1 \
+                    .conversations(conv_id) \
+                    .messages \
+                    .list(order='desc')
+        return self.messages
+
+def main():
+    args = parse_parameters()
+    if args.auth_token.lower() == 'secret':
+        args.auth_token = os.environ.get('TWIL_TOKEN')
+    if args.account_sid.lower() == 'secret':
+        args.account_sid = os.environ.get('TWIL_SID')
+    twilio = Twilio(args.auth_token, args.account_sid)
+    messages = twilio.get_conversation(args.conv_id, args.leng_limit)
+    messages_map = defaultdict(list)
+
+    # for msg in messages:
+    #     print(msg.date_created)
+    #     print(msg.author)
+    #     print(msg.body)
+
+    for msg in messages:
+        messages_map['date_created'].append(msg.date_created)
+        messages_map['author'].append(msg.author)
+        messages_map['body'].append(msg.body)
+
+    df = pd.DataFrame(messages_map)
+    df.dropna(inplace=True)
+    df.to_csv(args.local_dir+'/'+args.file_name, index=False)
+
+    # Store twilio csv as cnvrg dataset
+    if args.cnvrg_dataset.lower() != 'none':
+        cnvrg = Cnvrg()
+        ds = cnvrg.datasets.get(args.cnvrg_dataset)
+        try:
+            ds.reload()
+        except:
```

Comment:
Try to catch a specific exception

Reply:
Added custom NoneCnvrgDatasetError exception